axelor-contract/src/main/resources/views/Contract.xml

```diff
@@ -244,6 +245,9 @@
         <field name="currentContractVersion" expr="action:action-record-contract-version-default-record"/>
         <field name="partner" expr="eval: _partner"/>
         <field name="currency" expr="eval: __user__.activeCompany.currency"/>
+        <field name="targetTypeSelect" expr="eval:_customerContract" if="_customerContract != null"/>
+        <field name="targetTypeSelect" expr="eval:_supplierContract" if="_supplierContract != null"/>
```

Comment:
The 'targetTypeSelect' can be set with one context variable, not required separate for supplier and customer